### PR TITLE
ci: Automate the process to create a GitHub Release page

### PR DESCRIPTION
**Pull request by KengoTODA**
Currently the GitHub Release is created manually, so we sometimes have mis-operation like #755.
To avoid it, it's nice to automate the process to create a GitHub Release page for each release.

It is also good to list checksum of each artifact in the page, then user can [verify integrity of our products](https://docs.gradle.org/current/userguide/dependency_verification.html#sec:checksum-verification).

~TBD: Travis CI also creates GitHub Release, so to avoid conflict, need to move the build for Eclipse Plugin from Travis to GitHub Actions. refs #1123~ done.

### Sample output of the `createReleaseBody` task

You can preview this markdown at [this Gist](https://gist.github.com/KengoTODA/8a85542eeb9b569a45f35d1e4bf3d014).

```md
SpotBugs 4.0.4-SNAPSHOT

### CHANGELOG
- https://github.com/spotbugs/spotbugs/blob/4.0.4-SNAPSHOT/CHANGELOG.md

### CHECKSUM
| file | checksum (sha256) |
| ---- | ----------------- |
| spotbugs-4.0.4-SNAPSHOT-javadoc.jar | eff11a57457380d970f54a039f0b81b000733ea3c5e7767840211eee36ea0cbc |
| spotbugs-4.0.4-SNAPSHOT-sources.jar | 37d2fe5ce44a422da43e815fdd2488509e34ba7dfbf5c7358c9e7a9ddfa201a0 |
| spotbugs-4.0.4-SNAPSHOT.tgz | 3133a4a5e2480f6028559ccff209b70c032a8fb992b7a01a3b51b8e56df696c5 |
| spotbugs-4.0.4-SNAPSHOT.zip | 8f93b8f14d8a11ca3a2a050bd023cf1a8687eea06a968d463531d62cea7284dd |
| spotbugs-annotations-4.0.4-SNAPSHOT-javadoc.jar | 3b29e47c15d19d64bc34187d17c164a9ddc311723e06a5096ae5028e076d184c |
| spotbugs-annotations-4.0.4-SNAPSHOT-sources.jar | b338136e3e82d585348cde58a8fe3a678e16f51a35c31c1463e05fefef557aad |
| spotbugs-annotations.jar | acde8633c279c7a6d539b4aee5158d4cf24792f62477808134d95e967ad91af2 |
| spotbugs-ant-4.0.4-SNAPSHOT-javadoc.jar | d7330303e5098c893efe49a428ec5b323eb2b7987d0f0d5448600c0f05a9874d |
| spotbugs-ant-4.0.4-SNAPSHOT-sources.jar | c74dec42c0ed0dd1ae02a7410d8e0f0dbbee23e8e7da4a21910863677fcdbc8e |
| spotbugs-ant.jar | 9233e48d37882ae4e7a42e9f42ef4c63d6f802cf8f3b03ba575bee26e5032367 |
| spotbugs.jar | a3fa8545aff884fb883c20d4594613955f0ce823a865bdac983cb387462333d5 |
| test-harness-4.0.4-SNAPSHOT-javadoc.jar | 705d951e58e31f33a72fdb6f8046b1a7c70c09cbf999b4113bc17d611d286710 |
| test-harness-4.0.4-SNAPSHOT-sources.jar | 2c1f5ef929453f3b682c7eb7c1e22db3082b5f74c5a5be439be5dc31dd7a31aa |
| test-harness-4.0.4-SNAPSHOT.jar | 55d3a590b81ffec48293a76c45c0695914b405bf9f02bfb930e3ab99b5867d4f |
| test-harness-core-4.0.4-SNAPSHOT-javadoc.jar | a64b33e6fc29f91d670551b006f218de5fc1f51baa496152b25967db13b54ef4 |
| test-harness-core-4.0.4-SNAPSHOT-sources.jar | f320f5eb4069e9686b760b2a6a0760989753225f9e9ce1226e3258ec64795d8a |
| test-harness-core-4.0.4-SNAPSHOT.jar | cbec03867e077079d011e85f9932fb230fae3d909f741cffaa4c8097e91fdf40 |
| test-harness-jupiter-4.0.4-SNAPSHOT-javadoc.jar | 5817d35b07303cfd0ab35eb496adfdeac5670a45131737c55aa53987096907df |
| test-harness-jupiter-4.0.4-SNAPSHOT-sources.jar | 210353a57016e26b1a654d936a15f039613fa1ac532d485c1b1d03902f6c6315 |
| test-harness-jupiter-4.0.4-SNAPSHOT.jar | 17e8d78d1868f86e63f3e5e3d878e86f3d7fb1b8cf1a8d5f893333c982bfd3e2 |
```

### CODE REVIEW

**Comment by KengoTODA** [comment]
@spotbugs/everyone this PR is ready to merge, please have a review.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
           echo ossrhUsername=eller86 >> gradle.properties
           echo "ossrhPassword=${SONATYPE_PASSWORD}" >> gradle.properties
           echo "keystorepass=${KEYSTORE_PASS}" >> gradle.properties
-          ./gradlew publish --no-daemon
+          ./gradlew assemble publish --no-daemon --parallel
```
Comment (henrik242) [inline .github/workflows/release.yml:35]: Doesn't `publish` imply `assemble`? No biggie :)

Comment (KengoTODA) [reply .github/workflows/release.yml:35]: Yes, Eclipse plugin and site aren't targeted to `publish`, so we need to invoke `assemble` explicitly.

Comment (KengoTODA) [reply .github/workflows/release.yml:35]: Nice catch :)

**Pull request merged**